packages/wp-now/src/config/providers/WpEnvConfigProvider.ts

```diff
@@ -0,0 +1,99 @@
+import { SupportedPHPVersion } from '@php-wasm/universal';
+import * as fs from 'fs';
+import path from 'path';
+import { CliOptions } from './CliConfigProvider';
+import { WPEnvOptions } from './ConfigProvider';
+import { ConfigProviderInterface, WPNowOptions } from './ConfigProvider';
+
+const WP_ENV_FILE = '.wp-env.json';
+const WP_ENV_OVERRIDE_FILE = '.wp-env.override.json';
+
+class WpEnvConfigProvider implements ConfigProviderInterface {
```

Comment:
Something's off to me with the complexity here. I understand this class as a device to:

1. Read .wp-env.json and .wp-env.override.json
2. Convert them to WP now format
3. Merge the converted WP now configurations

We don't need priorities and complex patterns to do that. It could be as simple as three functions:

1. `parseWpEnvConfig( path: string ): WPNowOptions`
2. `mergeWpEnvConfigs( base: WPNowOptions, override: Partial<WPNowOptions> ): WPNowOptions`
3. `fromWpEnv( path: string ): WPNowOptions`

The one responsible for merging could even hardcode the properties like:

```js
return {
    php: override.php || base.php
}
```

Since merging object-based props and blueprints may require custom logic anyway.

Reply:
The priority is just to determine what config will be merged first. In our case this is:

- Arguments from terminal
- Wp env configs (wp-env.json and override)
- Defaults

I will change that, and remove the classes to keep it simple for now.

A file with some functions is sufficient for our use case I guess.

Reply:
Thanks man!